scripts/tcl_commands/routing.tcl

```diff
@@ -292,45 +254,57 @@ proc ins_diode_cells_4 {args} {
+    set save_def [index_file $::env(routing_tmpfiles)/diodes.def]
+    set save_odb [index_file $::env(routing_tmpfiles)/diodes.odb]
+
+    manipulate_layout $::env(SCRIPTS_DIR)/odbpy/diodes.py place\
+        -indexed_log [index_file $::env(routing_logs)/diodes.log]\
+        -output $save_odb\
+        -output_def $save_def\
         --diode-cell $::env(DIODE_CELL)\
         --diode-pin  $::env(DIODE_CELL_PIN)\
-        --fake-diode-cell $::antenna_cell_name\
-        $::env(CURRENT_DEF) |& tee $::env(TERMINAL_OUTPUT) $log
+        --fake-diode-cell $::antenna_cell_name
 
-    set_def $::env(SAVE_DEF)
+    set_def $save_def
+    set_odb $save_odb
 
     # Legalize
-    detailed_placement_or -def $::env(CURRENT_DEF) -log $::env(routing_logs)/diode_legalization.log
-
-    # Update netlist
-    write_verilog $::env(routing_results)/$::env(DESIGN_NAME)_diodes.v -log $::env(routing_logs)/write_verilog.with_diodes.log
-    set_netlist $::env(routing_results)/$::env(DESIGN_NAME)_diodes.v
+    detailed_placement_or\
+        -outdir $::env(routing_tmpfiles)\
+        -log $::env(routing_logs)/diode_legalization.log\
+        -name diodes
 
     TIMER::timer_stop
     exec echo "[TIMER::get_runtime]" | python3 $::env(SCRIPTS_DIR)/write_runtime.py "diode insertion - openlane"
-    if { $::env(LEC_ENABLE) } {
-        logic_equiv_check -rhs $::env(PREV_NETLIST) -lhs $::env(CURRENT_NETLIST)
-    }
+
+    # Update netlist
+    write_verilog $::env(routing_results)/$::env(DESIGN_NAME)_diodes.nl.v\
+        -powered_to $::env(routing_results)/$::env(DESIGN_NAME)_diodes.pnl.v\
+        -log $::env(routing_logs)/write_verilog.with_diodes.log
 }
 
 proc apply_route_obs {args} {
-    puts_info "Adding routing obstructions..."
-    # keep a warning for a while
-    puts_warn "Specifying a routing obstruction is now done using the coordinates"
-    puts_warn "of its bounding box instead of the now deprecated (x, y, size_x, size_y)."
-
-    try_catch $::env(OPENROAD_BIN) -python $::env(SCRIPTS_DIR)/odbpy/defutil.py add_def_obstructions\
-        --output [file rootname $::env(CURRENT_DEF)].obs.def \
-        --input-lef $::env(MERGED_LEF) \
-        --obstructions $::env(GRT_OBS) \
-        $::env(CURRENT_DEF) |& tee $::env(TERMINAL_OUTPUT) $::env(routing_logs)/obs.log
-
-    puts_info "Obstructions added over $::env(GRT_OBS)."
-    set_def [file rootname $::env(CURRENT_DEF)].obs.def
+    increment_index
+    TIMER::timer_start
+    set log [index_file $::env(routing_logs)/obs.log]
+    puts_info "Running Diode Insertion (log: [relpath . $log])..."
+
+    set save_def [file rootname $::env(CURRENT_DEF)].obs.def
+    set save_db [file rootname $::env(CURRENT_DEF)].obs.odb
+
+    manipulate_layout $::env(SCRIPTS_DIR)/odbpy/defutil.py add_obstructions\
+        -indexed_log $log\
+        -output $save_db\
+        -output_def $save_def\
+        --obstructions $::env(GRT_OBS)
+
+    puts_verbose "Obstructions added over $::env(GRT_OBS)."
+
+    set_def $save_def
```

Comment:
not that this is an issue, but for the sake of better understanding set_def isn't really neded here right?. it should have been handled in `manipulate_layout` just above it?

Reply:
No- manipulate_layout doesn't call `set_def` or `set_db`.